doc/hardware/porting/shields.rst

```diff
@@ -8,6 +8,282 @@ to extend its features and services for easier and modularized prototyping.
 In Zephyr, the shield feature provides Zephyr-formatted shield
 descriptions for easier compatibility with applications.
 
+Shield activation
+*****************
+
+Activate support for one or more shields by adding the matching ``--shield`` arguments
+to the west command:
+
+  .. zephyr-app-commands::
+     :app: your_app
+     :board: your_board
+     :shield: x_nucleo_idb05a1,x_nucleo_iks01a1
+     :goals: build
+
+
+Alternatively, it could be set by default in a project's CMakeLists.txt:
+
+.. code-block:: cmake
+
+	set(SHIELD x_nucleo_iks01a1)
+
+
+Shield hardware interfaces (form factors)
+*****************************************
+
+Shields are described by their mechanical connectors, and the electrical interface of the
+signals that are routed via the connectors.
+
+Mapping of the connector pins to the GPIOs of the board's microcontroller is done via a GPIO
+nexus node in the devicetree file of the board. The same node is used by the shield
+overlay file, if the shield uses GPIO functionality.
+Also electrical buses which are routed via the connector have devicetree node labels, for example
+I2C, UART and SPI buses.
+The board's devicetree file and the shield's overlay file are used together at build time.
+
+Zephyr supports a number of shield interfaces, one of them is the well-known Arduino
+form factor.
+
+For example a board without an accelerometer, but having an Arduino connector, can
+use an accelerometer via an Arduino shield. The devicetree file of the board defines
+an ``arduino_i2c`` node label for the I2C bus which is exposed to the Arduino connector.
+An accelerometer shield with Arduino connector will also use the ``arduino_i2c`` node label
+to describe how the accelerometer is connected. This is done in the ``.overlay`` file of the
+shield.
+
+Note that some boards and shields only support a subset of the features of a shield
+hardware interface.
+
+
+Arduino Uno R3
```

Comment:
can you sort all of these sections alphabetically please?

Reply:
Done